submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/helpers/SourceGenerator.kt

```diff
@@ -19,8 +19,19 @@ class SourceGenerator(
 ) {
     fun userSources(
         user: SecurityUser,
-        rootPath: String? = null
-    ): FilesSource = ComposedFileSource(userSourcesList(user, rootPath.orEmpty()))
+        rootPath: String? = null,
+        onBehalfUser: SecurityUser?
+    ): FilesSource {
```

Comment:
I don't think we should alter this method but instead, create the ComposedSource wherever required. This method is intended to create a source for any single user, not handling the `onBehalf` logic